services/director-v2/tests/unit/with_dbs/test_api_route_computations.py

```diff
@@ -313,6 +378,229 @@ async def test_create_computation(
     assert response.status_code == status.HTTP_201_CREATED, response.text
 
 
+@pytest.fixture
+def wallet_info(faker: Faker) -> WalletInfo:
+    return WalletInfo(wallet_id=faker.pyint(), wallet_name=faker.name())
+
+
+@pytest.fixture
+def fake_ec2_cpus() -> PositiveInt:
+    return 4
+
+
+@pytest.fixture
+def fake_ec2_ram() -> ByteSize:
+    return parse_obj_as(ByteSize, "4GiB")
+
+
+@pytest.fixture
+def mocked_clusters_keeper_service_get_instance_type_details(
+    mocker: MockerFixture,
+    default_pricing_plan_aws_ec2_type: str,
+    fake_ec2_cpus: PositiveInt,
+    fake_ec2_ram: ByteSize,
+) -> mock.Mock:
+    return mocker.patch(
+        "simcore_service_director_v2.modules.db.repositories.comp_tasks._utils.get_instance_type_details",
+        return_value=[
+            EC2InstanceType(
+                name=default_pricing_plan_aws_ec2_type,
+                cpus=fake_ec2_cpus,
+                ram=fake_ec2_ram,
+            )
+        ],
+    )
+
+
+@pytest.fixture
+def mocked_clusters_keeper_service_get_instance_type_details_with_invalid_name(
+    mocker: MockerFixture,
+    faker: Faker,
+    fake_ec2_cpus: PositiveInt,
+    fake_ec2_ram: ByteSize,
+) -> mock.Mock:
+    return mocker.patch(
+        "simcore_service_director_v2.modules.db.repositories.comp_tasks._utils.get_instance_type_details",
+        return_value=[
+            EC2InstanceType(
+                name=faker.pystr(),
+                cpus=fake_ec2_cpus,
+                ram=fake_ec2_ram,
+            )
+        ],
+    )
+
+
+@pytest.fixture(params=ServiceResourcesDictHelpers.Config.schema_extra["examples"])
+def project_nodes_overrides(request: pytest.FixtureRequest) -> dict[str, Any]:
+    return request.param
+
+
+async def test_create_computation_with_wallet(
+    minimal_configuration: None,
+    mocked_director_service_fcts: respx.MockRouter,
+    mocked_catalog_service_fcts: respx.MockRouter,
+    mocked_resource_usage_tracker_service_fcts: respx.MockRouter,
+    mocked_clusters_keeper_service_get_instance_type_details: mock.Mock,
+    product_name: str,
+    fake_workbench_without_outputs: dict[str, Any],
+    registered_user: Callable[..., dict[str, Any]],
+    project: Callable[..., Awaitable[ProjectAtDB]],
+    async_client: httpx.AsyncClient,
+    wallet_info: WalletInfo,
+    project_nodes_overrides: dict[str, Any],
+    default_pricing_plan_aws_ec2_type: str | None,
+    aiopg_engine: aiopg.sa.Engine,
+    fake_ec2_cpus: PositiveInt,
+    fake_ec2_ram: ByteSize,
+):
+    # In billable product a wallet is passed, with a selected pricing plan
+    # the pricing plan contains information about the hardware that should be used
+    # this will then override the original service resources
+    user = registered_user()
```

Comment:
TIP: when I repeat many times code in a test i tend to put it in a fixture. I wonder if 
```python
registered_user()
project()
```
could be refactored into two  interdependent fixtures

```python
@pytest.fixture
def user( create_register_user: Callable ):
     return create_regieter_user()
@pytest.fixture
async def project(create_project, user):
     return await create_project(user,  ... )


async def test_it( project ):
 ... 

```

Reply:
yeah well. it was not the goal of this PR. but of course.

Reply:
so same as with your other comment, this fixture comes from deep down the chain. I am not changing it here.